src/components/Map/Map.tsx

```diff
@@ -54,9 +54,9 @@ function restrictMovement(
 class Map extends React.Component<MapProps> {
   mapRef = React.createRef<HTMLDivElement>();
 
-  // State contains the instance of the HERE map to display
   state = {
-    map: null as H.Map | null,
+    map: null as H.Map | null, // Instance of the HERE map to display
+    clickedSensor: null as H.map.Marker | null, // Clicked marker
```

Comment:
nit: change variable name to `selectedSensor`, and references to this sensor from `clicked` to `selected`